src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -0,0 +1,1497 @@
+                    InitializeDebugTargetsCore(profiles);
+                });
+
+                var profileProvider = GetDebugProfileProvider();
+                _debugProfileProviderLink = profileProvider.SourceBlock.LinkTo(
+                    debugProfilesBlock,
+                    linkOptions: new DataflowLinkOptions { PropagateCompletion = true });
+            }
+        }
+        
+        public async override System.Threading.Tasks.Task Initialize()
+        {
+            // Need to set whether this is a web project or not since other parts of the code will use the cached value
+            await IsWebProjectAsync().ConfigureAwait(false);
+                        
+            // Create the debug targets dropdown
+            InitializeDebugTargets();
+        }
+
+        /// <summary>
+        /// Called whenever new settings are retrieved. Note that the controls which are affected depend heavily on the
+        /// currently selected profile.
+        /// </summary>
+        private void InitializeIISSettings(IIISSettings iisSettings)
+        {
+            if (iisSettings == null)
+            {
+                _currentIISSettings = null;
+                return;
+            }
+
+            _currentIISSettings = IISSettings.FromIIISSettings(iisSettings);
+
+            OnPropertyChanged(nameof(ApplicationUrl));
+            OnPropertyChanged(nameof(SSLEnabled));
+            OnPropertyChanged(nameof(SSLUrl));
+            OnPropertyChanged(nameof(WindowsAuthenticationEnabled));
+            OnPropertyChanged(nameof(AnonymousAuthenticationEnabled));
+        }
+
+        /// <summary>
+        /// Called when then the user saves the form.
+        /// </summary>
+        public async override Task<int> Save()
+        {
+
+            // For web projects, we need to validate the settings -especially the appUrl from which everything else hangs.
+            /* Uncomment the validation once IISsettings is working
+             * if (ContainsProfileKind(ProfileKind.IISExpress))
+            {
+                if (_currentIISSettings == null || _currentIISSettings.IISExpressBindingData == null || string.IsNullOrEmpty(_currentIISSettings.IISExpressBindingData.ApplicationUrl))
+                {
+                    throw new Exception(PropertyPageResources.IISExpressMissingAppUrl);
+                }
+                try
+                {
+                    Uri appUri = new Uri(_currentIISSettings.IISExpressBindingData.ApplicationUrl, UriKind.Absolute);
+                    if (appUri.Port < 1024)
+                    {
+                        throw new Exception(PropertyPageResources.AdminRequiredForPort);
+                    }
+                }
+                catch (UriFormatException ex)
+                {
+                    throw new Exception(string.Format(PropertyPageResources.InvalidIISExpressAppUrl, ex.Message));
+                }
+            }*/
+            if (ContainsProfileKind(ProfileKind.IIS))
```

Comment:
> if (ContainsProfileKind(ProfileKind.IIS)) [](start = 11, length = 42)

Interesting you didn't comment this out as well. FWIS -we never shipped IIS support, only IIS Express